phplib/SmartyWrap.php

```diff
@@ -57,7 +57,7 @@ static function mergeResources($files, $type) {
 
     // generate the output file if it doesn't exist or if it's too old
     if (!file_exists($output) || (filemtime($output) < $maxTimestamp)) {
-      $tmpFile = tempnam('/tmp', 'merge_');
+      $tmpFile = @tempnam('/tmp', 'merge_');
```

Comment:
În  general e bine să nu suprimăm erorile/avertismentele decât când există un motiv. În acest caz, aș vrea să aflu dacă generarea unui nume de fișier întâmpină probleme.

Reply:
Ar trebui parametrizat în dex.conf, nu? Altfel mi se oprește mereu la generare. Posibil ar fi în php,ini, dar atunci ar fi ceva global și nu e de dorit.

Reply:
Dar din nou, nu mai bine reparăm motivul pentru care funcția `tempnam()` dă erori? Ce anume se întâmplă?

Reply:
tempnam(): file created in the system's temporary directory
Since this is just a notice and the function still works correctly, you can either place the "@" operator infront of the function call "@tempnam()" or set your personal temp path in your php.ini like this:
Asta îmi apărea mereu când modificam un fișier și Smarty recompila.

Reply:
Ah, m-am prins, tu ești în Windows, nu? Atunci hai să rezolvăm problema într-un mod portabil. Cred că merge dacă înlocuiești `'tmp'` cu `sys_get_temp_dir()`. Acel avertisment nu apare tot timpul, ci doar dacă `tempnam()` nu reușește să creeze fișierul în directorul indicat (`'tmp'`). Ceea ce în Windows, evident, nu poate.

Reply:
OK!

Reply:
Acum văd că în dex.conf.sample există un astfel de parametru al aplicației. (tempDir = /tmp) Ori mergem pe acesta, ori pe sys_get_temp_dir(). Am mai găsit 2 locuri unde era hardcodat /tmp ca parametru pentru tempnam. Mai sunt și alte php-uri unde nu este parametrizată calea. Acolo rămân așa? Visual.php de exemplu.

Reply:
Aș merge pe `sys_get_temp_dir()`, renunțând la opțiunea din `dex.conf`. Dar nu te simți obligat să faci asta. :-) Dacă prefer, deschide doar un tichet, să nu uităm!